stargz/remote/resolver.go

```diff
@@ -205,12 +207,14 @@ func (tr *transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	return resp, nil
 }
 
-func redirect(ctx context.Context, endpointURL string, tr http.RoundTripper) (url string, err error) {
+func redirect(ctx context.Context, blobURL string, tr http.RoundTripper) (url string, err error) {
 	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
 	defer cancel()
 
-	// We use GET request for GCR.
-	req, err := http.NewRequestWithContext(ctx, "GET", endpointURL, nil)
+	// We use GET request for redirect.
+	// gcr.io returns 200 on HEAD without Location header (2020).
+	// ghcr.io returns 200 on HEAD without Location header (2020).
```

Comment:
You said ghcr doesn't allow HEAD, no?

Reply:
Sorry for the confusing description.

`ghcr.io` supports HEAD but it doesn't return Location header that we want for redirection.
```console
$ curl --head -H "Authorization: ..." https://ghcr.io/v2/stargz-containers/python/blobs/sha256:c9340f06be1e5222045e58720d755a4054ce843b691b7c3e63b769c1b10a82c0
HTTP/1.1 200 OK
... but without Location header ...
```

The s3 backend (`github-production-container-registry.s3.amazonaws.com`) doesn't allow HEAD where we want to get the blob size.
```console
$ curl --head 'https://github-production-container-registry.s3.amazonaws.com/blobs/...
HTTP/1.1 403 Forbidden
```